core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -753,21 +751,25 @@ class HttpGoogleServicesDAO(
     } )
   }
 
-  override def getGenomicsOperation(jobId: String): Future[Option[JsObject]] = {
-
-    import spray.json._
-
+  override def getGenomicsOperation(jobId: String): Future[JsObject] = {
     val opId = s"operations/$jobId"
-
     val genomicsApi = new Genomics.Builder(httpTransport, jsonFactory, getGenomicsServiceAccountCredential).setApplicationName(appName).build()
     val operationRequest = genomicsApi.operations().get(opId)
 
-    retryWithRecoverWhen500orGoogleError[Option[JsObject]](() => {
-      // Google library returns a Map[String,AnyRef], but we don't care about understanding the response
-      // So, use Google's functionality to get the json string, then parse it back into a generic json object
-      Some(executeGoogleRequest(operationRequest).toPrettyString.parseJson.asJsObject)
-    }) {
-      case t: HttpResponseException if t.getStatusCode == StatusCodes.NotFound.intValue => None
+    // This predicate is true when it's a 500 or Google error but NOT a Google 404.
+    //
+    // This uses the "applicative builder" syntax from cats to combine 2 predicates
+    // (when500orGoogleError and whenGoogle404) using a binary operator (e.g.
+    // (Boolean, Boolean) => Boolean), and return a resulting predicate.
+    val predicate = (when500orGoogleError |@| whenGoogle404).map((a, b) => a && !b)
```

Comment:
I'll change `predicate`.

Not sure I understand entirely. The previous logic caught Google 404's and returned Rawls 404's (with a similar error message). The new behavior for the 404 case is essentially the same:
```
https://local.broadinstitute.org:20443/api/workspaces/broad-dsde-dev/rtitle-1826-test/genomics/operations/foo

{
  "statusCode": 404,
  "source": "rawls",
  "causes": [],
  "stackTrace": [],
  "message": "Job not found: \"foo\""
}
```
The difference with the new code is, if Google returns an error besides 404, before the Rawls API would return 500 with the actual Google error buried somewhere in the stack trace. Now, the Google response status and message are used in the Rawls response. The status + message are displayed to the user in a modal dialog, so I thought it would be more useful to propagate the Google error instead of just showing a generic 500.

But if you feel this is not ideal I can change it.